pkg/plugin/system/system_test.go

```diff
@@ -0,0 +1,252 @@
+/*
+Copyright © 2022 Aspect Build Systems Inc
+
+Not licensed for re-use.
+*/
+
+package system
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/gomega"
+	"github.com/spf13/cobra"
+
+	rootFlags "aspect.build/cli/pkg/aspect/root/flags"
+	"aspect.build/cli/pkg/aspecterrors"
+	"aspect.build/cli/pkg/ioutils"
+	plugin_mock "aspect.build/cli/pkg/plugin/sdk/v1alpha2/plugin/mock"
+)
+
+func TestPluginSystem(t *testing.T) {
+	t.Run("executes hooks in reverse order of interceptors", func(t *testing.T) {
+		g := NewGomegaWithT(t)
+		ctrl := gomock.NewController(t)
+		defer ctrl.Finish()
+
+		// Minimal streams
+		var stdout strings.Builder
+		streams := ioutils.Streams{Stdout: &stdout, Stderr: &stdout}
+
+		// empty ctx + args + command
+		ctx := context.Background()
+		args := []string{}
+		cmd := &cobra.Command{
+			Use: "HookOrderTest",
+		}
+		// Required flags for interceptor hooks
+		cmd.PersistentFlags().Bool(rootFlags.InteractiveFlagName, false, "")
+
+		plugin := plugin_mock.NewMockPlugin(ctrl)
+
+		// Expect the callbacks in reverse-order of execution
+		gomock.InOrder(
+			plugin.EXPECT().PostRunHook(gomock.Any(), gomock.Any()),
+			plugin.EXPECT().PostTestHook(gomock.Any(), gomock.Any()),
+			plugin.EXPECT().PostBuildHook(gomock.Any(), gomock.Any()),
+		)
+
+		ps := NewPluginSystem().(*pluginSystem)
+		ps.Configure(streams)
+		ps.plugins.insert(plugin)
```

Comment:
As a continuation of my previous comment, you also don't want to do this in the tests. Instead, you instantiate a `pluginSystem` with the mocks and rely on their behaviour to control what happens inside the `Configure`. E.g. you will want your `Parser` to return the `aspectplugins` that then populate the `ps.plugins`.

Reply:
I'd rather not depend on anything in `Configure` though. The `commandHooksInterceptor` method really doesn't need anything from there other then a list of plugins.

One solution would be to essentially move the `commandHooksInterceptor` logic elsewhere so it can be unit tested without depending on things like `Configure`...